flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/sort/BinaryExternalSorter.java

```diff
@@ -1054,7 +1018,7 @@ private void releaseEmptyBuffers() {
 					}
 				}
 			}
-			releaseCoreSegments();
+			sortReadMemory.forEach(LazyMemorySegmentPool::cleanCache);
```

Comment:
We don't need to manually clean cache, as memory pool will try to clean cache when a segment is returned.